pkg/virtual/syncer/transformations/helpers.go

```diff
@@ -0,0 +1,107 @@
+/*
+Copyright 2022 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package transformations
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	"github.com/kcp-dev/kcp/pkg/apis/workload/v1alpha1"
+)
+
+// GetSyncing gathers, for each SyncTarget, all the information related
+// to the Syncing of the resource to this SyncTarget.
+// This information comes from labels and annotations.
+// Keys in the returns map are SyncTarget keys.
+func GetSyncing(upstreamResource metav1.Object) (map[string]SyncTargetSyncing, error) {
```

Comment:
I know, preexisting: `Syncing`. What is a syncing?

Reply:
The idea was simply to avoid the repetitive task of going through the various annotations / labels to get the Syncing intent for every SyncTarget.
Would it make sense to add this in the workload API helpers: https://github.com/davidfestal/kcp/blob/714c1b9b3fda1acfc1cab613e1365d1fd3877195/pkg/apis/workload/v1alpha1/helpers.go#L28 ?

Reply:
Moved to the new workload API `helpers` package.